enhancements/workload-partitioning/rationalizing-configuration.md

```diff
@@ -0,0 +1,478 @@
+   track that work.
+
+### User Stories
+
+#### Enabling and Configuring at the Same Time
+
+In this workflow, the user provides all of the information needed to
+fully partition the workloads from the very beginning.
+
+1. User runs the installer to `create manifests` to get standard
+   manifests without any partitioning.
+2. The user creates PerformanceProfile manifests for each known
+   MachineConfigPool and adds them to the installer inputs.
+3. The user runs the PAO `render` command to read the
+   PerformanceProfile manifests and create additional manifests, as
+   described above.
+4. User runs the installer to `create cluster`.
+5. Installer bootstraps the cluster.
+6. Bootstrapping runs the machine-config-operator (MCO) `render`
+   command, which generates MachineConfig manifests with low
+   precedence.
+7. Bootstrapping uploads both sets of MachineConfig manifests, one
+   after the other.
+8. The machine-config-operator (MCO) applies MachineConfigs to nodes
+   in the cluster.
+9. Bootstrapping finishes and the cluster is launched.
+10. If the MCO does not apply KubeletConfig in step 8, it must do it
+    here.
+11. The cluster has complete partitioning configuration for management
+    workloads.
+12. Kubelet starts and sees the config file enabling partitioning
+    (delivered in the MachineConfig manifest generated by
+    PAO). Kubelet advertises the workload resource type on the Node
+    and mutates static pods with partitioning annotations.
+13. The admission plugin uses the workload types on the new enablement
+    API to decide when to mutate pods.
+14. CRI-O sees pods with workload annotations and uses the resource
+    request to set cpushares and cpuset.
+15. On day 2 PAO, is installed and takes ownership of the
+    PerformanceProfile CRs.
+
+#### Enabling During Installation, Configuring Later
+
+In this workflow, the user provides enough information to *enable*
+workload partitioning but not enough to actually *configure* all nodes
+to partition the workloads into specific CPUs.
+
+1. User runs the installer to `create manifests` to get standard
+   manifests without any partitioning.
+2. The user runs the PAO `render` command without any
+   PerformanceProfile manifests to create the additional manifests, as
+   described above. The manifests for CRI-O do not include cpuset
+   configuration, because there are no PerformanceProfiles.
+3. User runs the installer to `create cluster`.
+4. Installer bootstraps the cluster.
+5. Bootstrapping runs the machine-config-operator (MCO) `render`
+   command, which generates MachineConfig manifests with low
+   precedence.
+6. Bootstrapping uploads both sets of MachineConfig manifests, one
+   after the other.
+7. The machine-config-operator (MCO) applies MachineConfigs to nodes
+   in the cluster.
+8. Bootstrapping finishes and the cluster is launched.
+9. If the MCO does not apply KubeletConfig in step 7, it must do it
```

Comment:
a little confused on this step 9, the cluster is up and done, but the MCO will just roll some changes out? what is the trigger on this? does the user have to do something?

Reply:
It wasn't clear to us whether the MCO render command consumed KubeletConfig resources, so that they would be included in the roll-out of the initial configuration to the host, or if that would have to happen once the cluster was up. From what @yuqi-zhang says above, it sounds like it's the latter, today. So the trigger is the MCO waking up and noticing that some of the extra manifests created by the PAO and added to the cluster during installation were KubeletConfig and that those haven't been applied.

The user can change the settings after the cluster is up using the PAO's PerformanceProfile API.

Reply:
I've added some wording to clarify the timing here.